### ISSUE TITLE
4 Phase so slow

### ISSUE THREAD

**Issue by kissaev** [madMAx43v3r/chia-plotter#320]
![Screenshot 2021-06-12 091632](https://user-images.githubusercontent.com/5376567/121767111-ff912a00-cb5e-11eb-9d0b-5143835821df.png)
![Screenshot 2021-06-12 091709](https://user-images.githubusercontent.com/5376567/121767116-0324b100-cb5f-11eb-9076-7296240a73c8.png)
32 threads of ryzen 5950x, 32 RAM@2800, m.2 nvme 970 evo plus 2TB, Windows 10
Same m.2 nvme 970 evo plus, without ramdisk, 4 phase always 500+ secs, but once i got 150 sec for phase 4. How is this possible and how to make it permanently? Can it be bug in windows build (0.4.0)?

**Comment by therealflinchy** [comment]
it's always about 160-180 for me in current linux build, 970 evo+ 1tb + ramdisk.

but other people get 35-50 seconds, with identical NVME and similar everything else. so there's something up still.

**Comment by madMAx43v3r** [comment]
yeah phase 4 still has many context switches, some CPUs don't handle it well

**Comment by kissaev** [comment]
> yeah phase 4 still has many context switches, some CPUs don't handle it well

![Screenshot 2021-06-12 143818](https://user-images.githubusercontent.com/5376567/121774679-3598d300-cb8c-11eb-90b7-d1d3134bd541.png)
Phase 4...

**Comment by madMAx43v3r** [comment]
that looks like a disk bottleneck

**Comment by therealflinchy** [comment]
phase 4 is slower for me with the latest/latest fork, was mostly 160-180s, up to 215ish now. not that it really matters too much when someone comes up with a manager/manager fork to stagger it at P4 though i guess.

**Comment by VertiHydro** [comment]
> phase 4 is slower for me with the latest/latest fork, was mostly 160-180s, up to 215ish now. not that it really matters too much when someone comes up with a manager/manager fork to stagger it at P4 though i guess.

Delucca has already opened a pull request for the Swar plot manager - you should be able to use that.
https://github.com/swar/Swar-Chia-Plot-Manager/pull/1198

**Comment by therealflinchy** [comment]
> > phase 4 is slower for me with the latest/latest fork, was mostly 160-180s, up to 215ish now. not that it really matters too much when someone comes up with a manager/manager fork to stagger it at P4 though i guess.
> 
> Delucca has already opened a pull request for the Swar plot manager - you should be able to use that.
> [swar/Swar-Chia-Plot-Manager#1198](https://github.com/swar/Swar-Chia-Plot-Manager/pull/1198)

awesome, thanks so much, just got it running, looking good, and logs!

**Comment by kissaev** [comment]
> that looks like a disk bottleneck

Then how it was 150 sec few times?

**Comment by kissaev** [comment]
> that looks like a disk bottleneck
It was windows/windows build bottleneck, on ubuntu i never see Phase 4 more then 50 sec :)
![image](https://user-images.githubusercontent.com/5376567/121884523-8fbba480-cd02-11eb-9ee4-aa48fc051916.png)

**Comment by nathandragun** [comment]
Hopefully this helps to substantiate something going on weird in Phase 4.  Attached is top -H screenshot showing what's happening during the writing for C1 and C3.  Disks are completely idle, threads are barely in use.

This specific Phase 4 took 293.385 sec

![Chia k32 plotter - 4846d0c (2990wx 128GB t 24 u 256 in Phase 4 C1 C3)](https://user-images.githubusercontent.com/40903109/122099643-3cb81f00-cde0-11eb-83d6-9d2318232e62.PNG)
Chia k32 plotter - 4846d0c (2990wx 128GB t 24 u 256 in Phase 4 C1 C3)
